### PR TITLE
DO NOT MERGE-configuration for runmodes 

### PR DESCRIPTION
**Pull request by DeepaliAgrwl**
Hi Rishabh, I have added the runmode configuration for qa and publish environment. Please have a look and let me know your comments

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OsgiConfig"
+    cartFetch.servicePath.string="/V1/guest-carts/"
+    productService.domainName="localhost/magento/rest"
+    productService.searchCriteriaField="store_id"
+    productService.searchCriteriaValue="4"
+    productService.servicePath="/us/V1/products"/>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : can we remove /us ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OsgiConfig"
+    cartFetch.servicePath.string="/us/V1/guest-carts/"
+    productService.domainName="localhost:8081/magento/rest"
+    productService.searchCriteriaField="store_id"
+    productService.searchCriteriaValue="4"
+    productService.servicePath="/us/V1/products"/>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : can we remove /us ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml:1]: @DeepaliAgrwl : why havent you added configuration for config.author.dev ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/org.apache.sling.commons.log.LogManager.factory.config-hclecomm.xml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OsgiConfig"
+    org.apache.sling.commons.log.file="logs/project-hclecomm.log"
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/org.apache.sling.commons.log.LogManager.factory.config-hclecomm.xml]: @DeepaliAgrwl : why is this config required ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OsgiConfig"
+    cartFetch.servicePath.string="/us/V1/guest-carts/"
+    productService.domainName="localhost:8081/magento/rest"
+    productService.searchCriteriaField="store_id"
+    productService.searchCriteriaValue="4"
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : what does searchCriteriaValue="4" signify ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OsgiConfig"
+    loginservice.domainName.string="localhost:8081/magento/rest"
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml]: @DeepaliAgrwl : why do we need port 8081 in publish ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OsgiConfig"
+    cartFetch.servicePath.string="/us/V1/guest-carts/"
+    productService.domainName="localhost:8081/magento/rest"
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : why do we need port 8081 in publish ?

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<jcr:root xmlns:sling="http://sling.apache.org/jcr/sling/1.0" xmlns:jcr="http://www.jcp.org/jcr/1.0"
+    jcr:primaryType="sling:OsgiConfig"
+    cartFetch.servicePath.string="/us/V1/guest-carts/"
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : can you pl remove /us

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml [hunk 9]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (rishabh-behl) [inline ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @DeepaliAgrwl : why do we have a separate runmode for publish ?

**Review by rishabh-behl — COMMENTED** [review]
@DeepaliAgrwl : all mandatory configuration should go inside /apps/config , any configuration specific to an instance will go inside author , publish and for environment we are supporting dev for now not QA or prod.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 5 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: Rishabh this is the store Id for the hclecomm store we created on magento

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 7 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: @rishabh-behl  we don't have publish environment setup now so provided the default values. we can change that later when we have separate publish environment

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 8 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: ok Rishabh

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 3 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml:1]: added

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/org.apache.sling.commons.log.LogManager.factory.config-hclecomm.xml (hunk 4 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/org.apache.sling.commons.log.LogManager.factory.config-hclecomm.xml]: deleted...

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 9 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: Removed

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 1 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.author.qa/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: Removed

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml (hunk 2 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.ProductServiceImpl.xml]: yes Rishabh. removed.

File: ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml (hunk 6 above)
Comment (DeepaliAgrwl) [reply ui.apps/src/main/content/jcr_root/apps/hclecomm/config.publish/com.hcl.ecomm.core.services.impl.LoginServiceImpl.xml]: removed Rishabh

**Pull request merged**